cfme/tests/storage/test_volume_snapshot.py

```diff
@@ -0,0 +1,174 @@
+        logger.warning("{name}:{msg} Volume deletion - skipping...".format(
+            name=type(e).__name__,
+            msg=str(e)))
+
+
+@pytest.yield_fixture(scope='function')
+def snapshot(volume):
+    # create new snapshot for crated volume
+    snapshot_name = fauxfactory.gen_alpha()
+    snapshot = volume.create_snapshot(snapshot_name)
+
+    try:
+        wait_for(lambda: snapshot.status == 'available',
+                 delay=20, timeout=1200, fail_func=snapshot.refresh)
+    except TimeoutException:
+        logger.error('Snapshot Creation fails:'
+                     'TimeoutException due to status not available (=error)')
+    yield snapshot
+
+    try:
+        if snapshot.exists:
+            snapshot.delete()
+    except Exception as e:
+        logger.warning("{name}:{msg}: Snapshot deletion - skipping...".format(
+            name=type(e).__name__,
+            msg=str(e)))
+
+
+@pytest.mark.tier(3)
+def test_storage_snapshot_create_cancelled_validation(volume):
+    """ Test snapshot create cancelled
+
+    prerequisites:
+        * Storage Volume
+
+    Steps:
+        * Navigate to Snapshot create window
+        * Fill snapshot name
+        * Click Cancel button
+        * Assert flash message
+    """
+
+    snapshot_name = fauxfactory.gen_alpha()
+    volume.create_snapshot(snapshot_name, cancel=True)
+    view = volume.create_view(VolumeDetailsView)
+    view.wait_displayed(timeout='10s')
+    view.flash.assert_message(
+        'Snapshot of Cloud Volume "{}" was cancelled by the user'.format(volume.name))
+
+
+@pytest.mark.tier(3)
+def test_storage_snapshot_create_reset_validation(volume):
+    """ Test snapshot create reset button validation
+
+    prerequisites:
+        * Storage Volume
+
+    Steps:
+        * Navigate to Snapshot create window
+        * Fill snapshot name
+        * Click Reset button
+        * Assert flash message
+    """
+
+    snapshot_name = fauxfactory.gen_alpha()
+    volume.create_snapshot(snapshot_name, reset=True)
+    view = volume.create_view(VolumeSnapshotView)
+    view.flash.assert_message('All changes have been reset')
+
+
+@pytest.mark.tier(1)
+def test_storage_volume_snapshot_crud(volume):
+    """ Test storage snapshot crud
+
+    prerequisites:
+        * Volume
+
+    Steps:
+        * Create a snapshot
+        * Delete a snapshot
+    """
+
+    # create new snapshot
+    initial_snapshot_count = volume.snapshots_count
+    snapshot_name = fauxfactory.gen_alpha()
+    snapshot = volume.create_snapshot(snapshot_name)
+    view = volume.create_view(VolumeDetailsView)
+    view.wait_displayed(timeout='10s')
+    view.flash.assert_success_message(
+        'Snapshot for Cloud Volume "{}" created'.format(volume.name))
+
+    # check for volume relationship tables snapshot count
+    try:
+        wait_for(lambda: volume.snapshots_count > initial_snapshot_count,
+                 delay=20, timeout=1000, fail_func=volume.refresh)
+    except TimeoutException:
```

Comment:
**Required**
This isn't the exception you want, this is `selenium.common.exceptions.TimeoutException`, which is specific to selenium timeouts.

`wait_for` raises a `TimedOutError`, which you can import from `cfme.utils.wait`.

Reply:
@mshriver added changes